### ISSUE TITLE
[Bug]: can't find a way to set default options ftb chunks/teams

### ISSUE THREAD

**Issue by Abalieno** [FTBTeam/FTB-Mods-Issues#921]
### Mod

FTB Teams

### Mod version

ftb-teams-fabric-1902.2.14-build.123

### Forge / Fabric version

19.2 14.22

### Modpack & version

mine

### What issue are you having?

I'm trying to set default options for "allow explosions" and "mob_griefing".

These are erroneously listed under ftb chunks options, but the corresponding options are MISSING from the server file at:
serverconfig/ftbchunks-world.snbt

These are only available under ftb teams for some reason, but I don't know any way to change default settings since a config file is unavailable.

That's all I'm trying to do, assigning default settings. I know how to do this for other ftb chunk settings, by creating overrides under "defaultconfigs". But since these settings don't appear in ftb chunks server options as they should, I'm at loss on how I'm able to change these options.

### Steps to reproduce

cannot set default ftb chunks options because erroneously assigned to ftb teams.

**Comment by desht** [comment]
They don't appear in server options because they're not server options - they're team-specific properties.  Which is why you change them in the team properties editor.  FTB Chunks registers these with FTB Teams when it starts up.

Having said that, it would be very useful to have defaults in the server config for these settings. I will consider how best to add this - it's likely to need changes to both Chunks and Team to support it, due to the way defaults are currently initialised (the order in which the FTB Chunks server config is loaded and team properties are registered needs to be fully predictable, which it isn't right now).

**Comment by Abalieno** [comment]
Would there be a way to workaround this using some kind of script/command to load automatically at server creation through kubejs?

Something like "event.server.runCommandSilent" that I could use with a trigger?

**Comment by desht** [comment]
Well, there is this command: `/ftbteams party settings_for <party_name> ftbchunks:allow_explosions false`, but that does need a party name (and that the team is a party team), so server start wouldn't be an option.

However, FTB Teams does fire a `TeamCreatedEvent` when a party is created, which should in theory be interceptable via KubeJS (but I'm not a KJS expert). The Java event is here: https://github.com/FTBTeam/FTB-Teams/blob/1.19/main/common/src/main/java/dev/ftb/mods/ftbteams/event/TeamCreatedEvent.java.  The team name as passed to the command can be extracted (in Java) with `event.getTeam().getStringId()` - hopefully you can figure out how to do this with KJS?

The limitation with this is it only works for party teams, not individual player teams.  I am working on making these team properties honour defaults from FTB Chunks config, but that will need new versions of both Chunks and Teams.  Hopefully within the next week or two.

**Comment by Abalieno** [comment]
We are a week or two into 2024, maybe there's still hope? :)

**Comment by MichaelHillcox** [comment]
Likely got missed off the list. I'll poke Des about this once they're back from holiday :+1:

**Comment by desht** [comment]
Sorry, this did fall by the wayside a bit.  Good news is that the necessary changes are in FTB Teams now (as of 2001.2.0 release), so getting support for team property defaults into the next FTB Chunks release shouldn't be a problem!

**Comment by Abalieno** [comment]
Well, I'm on 1.19/fabric, but nice job anyway, then.

**Comment by desht** [comment]
Yeah, sorry, this won't be getting backported to 1.19.  Only trivial and/or critical issues get backports, and this is neither.

**Comment by phit** [comment]
This would still be nice, there used to be settings to force these for teams via FTB Chunks configuration and FTB Ranks permissions too.. not a fan of just making a cobblestone box for wither containment and no way to turn it off server side.

**Comment by desht** [comment]
Thanks for the reminder. Getting this into 1.21.1 now, 1.20.1 to follow very shortly.

**Comment by phit** [comment]
yes still relevant

**Comment by Gaz492** [comment]
Please disregard the previous reply from the stale bot, it's been a bad bot and sent for recycling.

**Comment by MCChiter** [comment]
Soooo are there any news regarding this? i tried to set up something and well, there is a reason i am on github xD

**Comment by desht** [comment]
Sorry, it fell by the wayside for 1.20. I've backported the change now and will get a new 1.20 release out soon.

**Comment by phit** [comment]
is there some other issue open for setting fixed configs, forcing mob griefing to be enabled is the main use case I've been waiting for

**Comment by desht** [comment]
This is in the latest FTB Chunks release for 1.20.
